crates/matrix-sdk-ui/src/timeline/inner/state.rs

```diff
@@ -891,7 +888,7 @@ impl TimelineInnerMetadata {
         let Some(fully_read_event) = &self.fully_read_event else { return };
         trace!(?fully_read_event, "Updating read marker");
 
-        let read_marker_idx = find_read_marker(items);
+        let read_marker_idx = items.iter().rposition(|item| item.is_read_marker());
```

Comment:
Personally I prefer a named method telling me what this code does. Was there a specific reason you thought it was better inline?

Reply:
Interesting. So besides a previous background as a compiler engineer (which taught me that inlining is the mother of all optimizations), the most important reason is that I navigate a lot in the code by looking at the call tree: jump to definition of a function / looks at callers of a function. A function that's a one-liner, and is used only once, is an unnecessary detour, with this mindset. (Also I find that good function names are really hard to get right, when seeing the code really tells you what it does in a more concrete way.)

In this particular instance, I find that the inlined function's body is more telling than the function call (it shows me it's an optimistic reverse linear scan, which the function name doesn't show). Overall I prefer it a lot, but if many people are opposed, I can revert to just moving this function much closer to where it's used.

Reply:
My 2c: Either is fine, maybe slight preference for inlining as the definition is / was so trivial.

Reply:
Either way is fine with me, and if I saw the code as you propose, I wouldn't be arguing to factor it out.

I would say though that if naming it is hard, that might just be because it's _worth_ naming. (Of course, it might also be because the factoring is wrong i.e. it shouldn't be a separate function...)